src/flow_solver/flow_solver.cpp

```diff
@@ -42,7 +42,11 @@ FlowSolver<dim, nstate>::FlowSolver(
 , dg(DGFactory<dim,double>::create_discontinuous_galerkin(&all_param, poly_degree, flow_solver_param.max_poly_degree_for_adaptation, grid_degree, flow_solver_case->generate_grid()))
 {
     flow_solver_case->set_higher_order_grid(dg);
-    dg->allocate_system();
+    if (flow_solver_param.steady_state == false) {
```

Comment:
Both implicit and explicit methods can be used for steady state and unsteady problems. For example Implicit RK that @cpethrick is implementing. (I know she uses matrix-free, but we don't need to build the matrix because it's matrix-free, not because it's steady)

Reply:
Ah okay, so we should be checking the ODE solver type here instead?

Reply:
We could add a parameter `allocate_AD_matrices` which is set `true` according to the user's chosen `ode_solver_type` (rather than adding a parameter which the user would specify explicitly). Or just add some logic here if such a parameter isn't useful elsewhere.

Reply:
Maybe each ODE class should have a bool allocate_AD_matrices that gets assigned in the ODE solver base constructor; similar to how in Physics we have the bool has_nonzero_dissipative_flux. That way there's no dependency on FlowSolver

Reply:
Nvm, this won't work since we create the ODE solver after having allocated DG

Reply:
I'll go with a parameter approach, that gets assigned in the parameter reader based on the ODE solver type

Reply:
fixed